hw_8.py

```diff
@@ -0,0 +1,129 @@
+"""Homework_8"""
+
+# # # Последовательность # # #
+# Дана последовательность целых чисел в виде массива.
+# Определите, можно ли получить строго возрастающую последовательность,
+# удалив из массива не более одного элемента.
+# Примечание: последовательность a0, a1, ..., an считается строго возрастающей,
+# если a0 < a1 < ... < an.
+# Последовательность, содержащая только один элемент,
+# также считается строго возрастающей.
+# Примеры
+# Для последовательности = [1, 3, 2, 1], вывод должен быть решение = False.
+# В этом массиве нет ни одного элемента, который можно было бы удалить,
+# чтобы получить строго возрастающую последовательность.
+# Для последовательности = [1, 3, 2] вывод должен быть = True.
+# Вы можете удалить 3 из массива,
+# чтобы получить строго возрастающую последовательность [1, 2].
+# Альтернативно можно убрать 2,
+# чтобы получить строго возрастающую последовательность [1, 3].
+#
+# solution([1, 2, 3])
+# solution([1, 2, 1, 2])
+# solution([1, 3, 2, 1])
+# solution([1, 2, 3, 4, 5, 3, 5, 6])
+# solution([40, 50, 60, 10, 20, 30])
+
+
+def sequence_check(sequence):
+    """Проверка последовательности на возрастание"""
+    list_sequence = list(set(sequence))
+    if len(sequence) - len(list_sequence) >= 2:
+        return False
+
+    list_sequence.sort()
+
+    if len(sequence) - len(list_sequence) == 0:
+        delete_count = 0
+    else:
+        delete_count = 1
+
+    # Проверяем, можно ли получить строго возрастающую последовательность
+    for i in range(len(list_sequence)):
+        if sequence[i] != list_sequence[i]:
+            if sequence[i] < list_sequence[i]:
+                delete_count += 1
+                if delete_count > 1:
+                    return False
+            else:
+                if i == 0 or sequence[i] <= list_sequence[i - 1]:
+                    return False
+    return True
+
+
+sequence = [1, 3, 2, 1]
+print(sequence_check(sequence))
```

Comment:
Using `assert` here is better than `print()`

Reply:
fixed